report/src/index.ts

```diff
@@ -0,0 +1,242 @@
+import { QuartileCandlestick } from "@hpcc-js/chart";
+import { Widget } from "@hpcc-js/common";
+import { Result, Workunit, WUInfo } from "@hpcc-js/comms";
+import { Grid } from "@hpcc-js/layout";
+import { Html } from "@hpcc-js/other";
+import { BreakdownTable, StyledTable } from "@hpcc-js/html";
+
+const knownProfileField = (sch: WUInfo.ECLSchemaItem): boolean => ["attribute", "given_attribute_type", "best_attribute_type", "rec_count", "fill_count", "fill_rate", "cardinality", "cardinality_breakdown", "modes", "min_length", "max_length", "ave_length", "popular_patterns", "rare_patterns", "is_numeric", "numeric_min", "numeric_max", "numeric_mean", "numeric_std_dev", "numeric_lower_quartile", "numeric_median", "numeric_upper_quartile", "numeric_correlations"].indexOf(sch.ColumnName) > 0;
+const countProfileFields = (r: Result): number => r.ECLSchemas.ECLSchemaItem.filter(knownProfileField).length;
+const isProfileResult = (r: Result, threshold = 4): boolean => countProfileFields(r) >= threshold;
+
+const config = {
+    rowHeight: 180,
+    primaryFontSize: 20,
+    secondaryFontSize: 14,
+    primaryColor: "#494949",
+    secondaryColor: "#DDD",
+    offwhiteColor: "#FBFBFB",
+    blueColor: "#1A99D5"
+};
+
+export class Test extends Grid {
```

Comment:
`Test` should be `DataPatternsReport1` or something similar

Reply:
Go wild - this PR is a staring point for you...